lib/board.rb

```diff
@@ -0,0 +1,69 @@
+class Board
+
+  attr_accessor :cells
+
+
+  def initialize
+    self.cells ||= Array.new(9, " ")
+  end # initialize
+
+
+  ## displays the board
+  def display
+
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+
+  end # board
+
+
+  ## resets the array if necessary
+  def reset!
+    self.cells = Array.new(9, " ")
+  end # reset!
+
+
+  ## determines the value in the input position
+  def position(user_input)
+    cells[user_input.to_i - 1]
+  end # position
+
+
+  ## determines if the board is full
+  def full?
+    true if cells.all?{|pos| pos == "X" || pos == "O"}
```

Comment:
Avoid returning boolean literals from `if` statements, relying instead on the implicit return value of the boolean expression. This method's content could read: `cells.all?{|pos| pos == "X" || pos == "O"}`